src/test/java/htsjdk/samtools/cram/CRAMCodecCorpusTest.java

```diff
@@ -0,0 +1,343 @@
+        // concatenate RANS4x8 and RANSNx16 codecs and testdata
+        return Stream.concat(Arrays.stream(getRANS4x8TestData()), Arrays.stream(getRANS4x16TestData()))
+                .toArray(Object[][]::new);
+    }
+
+    @Test (
+            dataProvider = "allRansCodecsAndData",
+            dependsOnMethods = "testGetHTSCodecsCorpus",
+            description = "Roundtrip using htsjdk RANS. Compare the output with the original file" )
+    public void testRANSRoundTrip(
+            final Path inputTestDataPath,
+            final RANSEncode ransEncode,
+            final RANSDecode ransDecode,
+            final RANSParams params,
+            final String unusedCompressedDirname) throws IOException {
+        if (!CRAMCodecCorpus.isHtsCodecsTestDataAvailable()) {
+            throw new SkipException("htscodecs test data is not available locally");
+        }
+        try (final InputStream is = Files.newInputStream(inputTestDataPath)) {
+
+            // preprocess the uncompressed data (to match what the htscodecs-library test harness does)
+            // by filtering out the embedded newlines, and then round trip through RANS and compare the
+            // results
+            final ByteBuffer uncompressedBytes = ByteBuffer.wrap(filterEmbeddedNewlines(IOUtils.toByteArray(is)));
+            if ((params.getFormatFlags() & STRIPE_FLAG_MASK)!=0) {
+
+                // If Stripe Flag is set, skip the round trip test as encoding is not implemented for this case.
+                // TODO: Assert raise Exception
+                System.out.printf("Stripe Flag is set. Skipping testRANSRoundTrip for file: %s. " +
+                        "Format Flags: %s . The current RANSNx16 implementation does not " +
+                        "support encoding when Stripe Flag is set%n", inputTestDataPath.toString(), params.getFormatFlags());
+            }
+            else {
+                final ByteBuffer compressedBytes = ransEncode.compress(uncompressedBytes, params);
+                uncompressedBytes.rewind();
+                System.out.printf("filename:%s %s Uncompressed: (%,d) Compressed: (%,d)%n",
+                        inputTestDataPath.getFileName(),
+                        params.toString(),
+                        uncompressedBytes.remaining(),
+                        compressedBytes.remaining());
+                Assert.assertEquals(ransDecode.uncompress(compressedBytes), uncompressedBytes);
+            }
+        }
+    }
+
+    @Test (
+            dataProvider = "allRansCodecsAndData",
+            dependsOnMethods = "testGetHTSCodecsCorpus",
+            description = "Compress the original file using htsjdk RANS and compare it with the existing compressed file. " +
+                    "Uncompress the existing compressed file using htsjdk RANS and compare it with the original file.")
+    public void testRANSPreCompressed(
+            final Path inputTestDataPath,
```

Comment:
Its a bit hard to derive what many of these paths, files and streams represent when they have ambiguous names like this (I know I probably wrote some of them, but on reviewing this I'm finding it hard to keep track). It would be helpful to do a pass over all the variable names in this file, and use a canonical naming format, i.e, a `compressed` or `uncompressed` prefix, followed by `htsjdk` or `corpus` depending on who created it,  followed by `Path` or `Stream`, depending on what it is. So this one would be `uncompressedCorpusPath` ??. 

This file is likely to get a lot larger as we add more codecs (or maybe they should be broken up, renaming this one to `RANSCorpusTest`) ?

Reply:
done